### ISSUE TITLE
consider exposing kubernetes services across the cluster

### ISSUE THREAD

**Issue by ryane** [CiscoCloud/mantl#1728]
Right now, kubernetes services can be reached from control nodes and kubeworker nodes but cannot be accessed from any other node type.

Example service:

```shell
$ kubectl run service-test --image=nginx
deployment "service-test" created

$ kubectl expose deployment service-test --type=NodePort --port=80
service "service-test" exposed

$ kubectl describe svc service-test
Name:                   service-test
Namespace:              default
Labels:                 run=service-test
Selector:               run=service-test
Type:                   NodePort
IP:                     10.254.189.112
Port:                   <unset> 80/TCP
NodePort:               <unset> 31929/TCP
Endpoints:              192.168.1.2:80
Session Affinity:       None
No events.
```

From a control node:

```shell
# cluster ip
$ curl -sI 10.254.189.112 | head -n 1
HTTP/1.1 200 OK

# pod endpoint
$ curl -sI 192.168.1.2:80 | head -n 1
HTTP/1.1 200 OK

# node port
$ curl -sI $HOSTNAME:31929 | head -n 1
HTTP/1.1 200 OK
```

All of the above are reachable via kubeworkers as well. However, from worker or edge nodes these endpoints are unavailable. This means applications running on workers (via mesos or otherwise) cannot communicate with apps running on kubernetes.

A simple way to enable this is to install the kubernetes components (`kubernetes` and `kubernetes-node` roles) on all nodes but set `--register-schedulable=false` on the kubelet for all nodes except kubeworkers. With this, k8s workloads will only be scheduled on kubeworkers but kube-proxy on every node will set up the iptables rules to enable connectivity.  It may be possible to just install a subset of the kubernetes components (just kube-proxy maybe?) but more investigation would be needed.

**Comment by ryane** [comment]
it  does look like we would just need to run kube-proxy on worker and edge nodes to enable this.

**Comment by siddharthist** [comment]
I'm definitely for this, no reason we shouldn't.

**Comment by thomasvincent** [comment]
register-schedulable is deprecated. We should not register-schedulable false to taint a node before registering it to a k8s cluster. For that use-case, register-with-taints can be used instead today.

**Comment by Cryptophobia** [comment]
Why is this important or useful? Is there a use case where someone is going to build a service or scheduler deployed through marathon that will work with kubernetes services... ?

**Comment by thomasvincent** [comment]
Yes, I could see one. Right now people wrote there own schedulers in Mesos. But I could see the use case for both.

**Comment by Theaxiom** [comment]
I don't see why not.

**Comment by KaGeN101** [comment]
This is our case. Pretty sure the way calico is setup now this is achieved? Containers in mesos should be on the same network as containers in k8, or that is what calico is suppose to do. They should behave like a single solution.

**Comment by KaGeN101** [comment]
Will verify will deploy but would definitely expect it to be this way

**Comment by Theaxiom** [comment]
@KaGeN101 You are correct, we just have to verify this is how it is currently working.
